lib/scholar/neighbors/kd_tree.ex

```diff
@@ -171,7 +181,7 @@ defmodule Scholar.Neighbors.KDTree do
       end
 
     k = rem(level, dims)
-    Nx.argsort(tensor[[.., k]] + band * tags, type: :u32)
+    Nx.argsort(tensor[[.., k]] + band * tags, type: :u32) |> Nx.as_type(:s64)
```

Comment:
This is a temporary change, because when I ran the code with `EXLA` I got the following type mismatch:

```
** (RuntimeError) Dynamic slice start indices must all have the same shape, got mismatching indices with shapes u32[] and s64[].
    (exla 0.7.0-dev) lib/exla/op.ex:876: EXLA.Op.unwrap!/1
    (exla 0.7.0-dev) lib/exla/op.ex:121: EXLA.Op.get_shape/1
    (exla 0.7.0-dev) lib/exla/defn.ex:1043: EXLA.Defn.to_operator/4
    (exla 0.7.0-dev) lib/exla/defn.ex:646: EXLA.Defn.cached_recur_operator/4
    (exla 0.7.0-dev) lib/exla/defn.ex:490: EXLA.Defn.recur_operator/3
    (elixir 1.15.5) lib/enum.ex:1819: Enum."-map_reduce/3-lists^mapfoldl/2-0-"/3
    (elixir 1.15.5) lib/enum.ex:1819: Enum."-map_reduce/3-lists^mapfoldl/2-0-"/3
    iex:10: (file)
```